### PR TITLE
Responsive navbar

### PR DESCRIPTION
**Pull request by ClaraChaouat**
Hi Sandrina, 

This is a first draft of the responsive navbar.

##Struggles
- button styling: when I try to change the button styling using media query, it doesnt work. So I thought It had something to do with the fact that I created a component for buttons and therefore I tried to use the media query in the button.scss file but it doesnt work neither.

- Images: I'd like to understand what are the best practices with image sizes when doing integration. Should we keep the image dimensions as there are on the figma file? Another question regarding images; I read different things about responsive images. But what are the best practices for images?

Thanks.

### CODE REVIEW

File: src/styles/responsive.scss [hunk 1]
```diff
@@ -0,0 +1 @@
+$mobile-version: "only screen and (max-width: calc(600 / 16 * 1rem))";
```
Comment (sandrina-p) [inline src/styles/responsive.scss:1]: question: Can you explain what this calc does and what's its benefit compared to normal `37.5rem`? And why rem instead of px? :)

P.S. It's well done, I'm just asking to double check if you understand what's happening here :)

File: src/components/button.scss [hunk 2]
```diff
@@ -19,3 +20,11 @@
 .btn--secondary {
   background-color: $secondaryColor;
 }
+
+@media #{$mobile-version} {
```
Comment (sandrina-p) [inline src/components/button.scss:24]: > button styling: when I try to change the button styling using media query, it doesnt work. So I thought It had something to do with the fact that I created a component for buttons and therefore I tried to use the media query in the button.scss file but it doesnt work neither.

What do you mean it doesn't work? It works on my side :)

image

File: src/components/hero.scss [hunk 3]
```diff
@@ -19,8 +19,9 @@
       font-style: normal;
       font-weight: 800;
       font-size: 64px;
-      line-height: 70px;
+      line-height: 1.1;
       flex-wrap: wrap;
+      max-width: 517px;
```
Comment (sandrina-p) [inline src/components/hero.scss:24]: question: What does 517px mean? It's good to leave a code comment explaining the way of this "magic" numbers to help other developers understand your code :)

File: src/components/navigationbar.scss [hunk 4]
```diff
@@ -65,3 +66,25 @@
...
+    width: 100%;
+  }
+
+  .navBarLogIn {
+    a {
+      font-size: 12px;
+    }
+
+    button {
+      background-color: red;
```
Comment (sandrina-p) [inline src/components/navigationbar.scss]: If you are talking about this one not working... it's because the CSS specificity level is higher than the mobile version. You can see it on the Devtools the rule ~~in strikethrough text~~. Do you understand how/why the specificity is lower?

image

File: src/components/navigationbar.scss [hunk 5]
```diff
@@ -65,3 +66,25 @@
     }
   }
 }
+
+@media #{$mobile-version} {
```
Comment (sandrina-p) [inline src/components/navigationbar.scss:70]: question: I noticed that in both changes you did not use "mobile first approach". Why?

File: src/components/Hero.js [hunk 6]
```diff
@@ -17,9 +17,7 @@ export function Hero() {
     <div className="subheader">
       <div className="subheaderTitle">
         <div className="subheaderTitleContainer">
-          <p>
-            Where hiring <br /> comes together.
-          </p>
+          <p>Where hiring comes together.</p>
           <img
```
Comment (sandrina-p) [inline src/components/Hero.js:21]: > Images: I'd like to understand what are the best practices with image sizes when doing integration. Should we keep the image dimensions as there are on the figma file? Another question regarding images; I read different things about responsive images. But what are the best practices for images?


Good practices depend from scenario to scenario. I don't have any good resource (article) that talks about it in a comprehensive way. I'd need a more specific question, so that I could give you a better answer.

Nevertheless, here's some generic guidelines:
- The image ratio itself should stay the same, regardless of the screen size. The image should not stretch (change its ratio), otherwise, it would look ugly.
- If the image is irrelevant, you might hide it on smaller screens to save space.
- Using `img { max-width: 100%; }` is a good global approach to ensure images don't grow wider than their original size (to avoid blury images)

File: src/styles/responsive.scss (hunk 1 above)
Comment (ClaraChaouat) [reply src/styles/responsive.scss:1]: @sandrina-p To answer your first question: calc allows us to do maths, here it allows us to convert px to rem.
Then, to answer you second question about why I used rem instead of px : 

with a 16px base font size, 68.75rem equals 1100px (68.75 * 16). If our window is 1200px, we'll get the desktop styles.
But, if the user cranks their default font size to 24px, the math changes. Now, this media query only triggers at 1650px (68.75 * 24). At 1200px, we're getting the tablet styles, not the desktop ones.
In other words, rem-based viewports scale with the user's chosen default font size. As the font size gets bigger, the breakpoints slide up the scale. (From Josh Commeau course)

File: src/components/hero.scss (hunk 3 above)
Comment (ClaraChaouat) [reply src/components/hero.scss:24]: @sandrina-p First, It is the size mentionned on the design. Then, it allows to break the sentence avoiding using <br>

File: src/components/navigationbar.scss (hunk 5 above)
Comment (ClaraChaouat) [reply src/components/navigationbar.scss:70]: From what I understand from the article you sent me, choosing the destock design or the mobile-first approach is something you do at the very beginning. Now that I did 95% of the job, I won't do rebuild the code from scratch. I'll do it for another project.

File: src/components/navigationbar.scss (hunk 4 above)
Comment (ClaraChaouat) [reply src/components/navigationbar.scss]: I just found out why. Thanks for the tip!
